ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionProcessor.java

```diff
@@ -208,193 +209,204 @@ public Result processTransaction(
+      senderMutableAccount.incrementBalance(refundedWei);
+
+      final Gas gasUsedByTransaction =
+          Gas.of(transaction.getGasLimit()).minus(initialFrame.getRemainingGas());
+
+      final MutableAccount coinbase = worldState.getOrCreate(miningBeneficiary).getMutable();
+      final Gas coinbaseFee = Gas.of(transaction.getGasLimit()).minus(refunded);
+      if (blockHeader.getBaseFee().isPresent() && transaction.isEIP1559Transaction()) {
+        final Wei baseFee = Wei.of(blockHeader.getBaseFee().get());
+        if (transactionGasPrice.compareTo(baseFee) < 0) {
+          return Result.failed(
+              gasUsedByTransaction.toLong(),
+              refunded.toLong(),
+              ValidationResult.invalid(
+                  TransactionValidator.TransactionInvalidReason.TRANSACTION_PRICE_TOO_LOW,
+                  "transaction price must be greater than base fee"),
+              Optional.empty());
+        }
+      }
+      final CoinbaseFeePriceCalculator coinbaseCreditService =
+          transaction.isFrontierTransaction()
+              ? CoinbaseFeePriceCalculator.frontier()
+              : coinbaseFeePriceCalculator;
+      final Wei coinbaseWeiDelta =
+          coinbaseCreditService.price(coinbaseFee, transactionGasPrice, blockHeader.getBaseFee());
 
-    initialFrame.getSelfDestructs().forEach(worldState::deleteAccount);
+      coinbase.incrementBalance(coinbaseWeiDelta);
 
-    if (clearEmptyAccounts) {
-      clearEmptyAccounts(worldState);
-    }
+      initialFrame.getSelfDestructs().forEach(worldState::deleteAccount);
 
-    if (initialFrame.getState() == MessageFrame.State.COMPLETED_SUCCESS) {
-      return Result.successful(
-          initialFrame.getLogs(),
-          gasUsedByTransaction.toLong(),
-          refunded.toLong(),
-          initialFrame.getOutputData(),
-          validationResult);
-    } else {
-      return Result.failed(
-          gasUsedByTransaction.toLong(),
-          refunded.toLong(),
-          validationResult,
-          initialFrame.getRevertReason());
+      if (clearEmptyAccounts) {
+        clearEmptyAccounts(worldState);
+      }
+
+      if (initialFrame.getState() == MessageFrame.State.COMPLETED_SUCCESS) {
+        return Result.successful(
+            initialFrame.getLogs(),
+            gasUsedByTransaction.toLong(),
+            refunded.toLong(),
+            initialFrame.getOutputData(),
+            validationResult);
+      } else {
+        return Result.failed(
+            gasUsedByTransaction.toLong(),
+            refunded.toLong(),
+            validationResult,
+            initialFrame.getRevertReason());
+      }
+    } catch (final RuntimeException re) {
+      LOG.error("Critical Exception Processing Transaction", re);
+      return Result.invalid(
+          ValidationResult.invalid(
+              TransactionInvalidReason.INTERNAL_ERROR,
+              "Internal Error in Besu - " + re.toString()));
     }
```

Comment:
How come we don't need this in the `PrivateTransactionProcessor`?

Reply:
We need it there too.  Added.